### PR TITLE
API for projects.json and project/{id}.json (first pass)

### PR DESCRIPTION
**Pull request by mikelmaron**
@pgiraud first pass at /projects.json and /project/{id}.json. Still more to do, but thought I'd start off the pull request discussion now, to check whether this is on the right track.

### CODE REVIEW

File: osmtm/views/project.py [hunk 1]
```diff
@@ -100,6 +100,34 @@ def project(request):
                 priority_areas=FeatureCollection(features),)
 
 
+@view_config(route_name="project_json", renderer='json',
+             permission="project_show",
+             http_cache=0)
+def project_json(request):
+    id = request.matchdict['project']
+    request.response.content_disposition = \
+        'attachment; filename="hot_osmtm_%s.json"' % id
```
Comment (pgiraud) [inline osmtm/views/project.py]: I would avoid putting HOT in the file name. The tasking manager can be used by other organizations.

**Comment by mikelmaron** [comment]
@pgiraud I updated the filenames for json download (hadn't changed that from prior implementation).

**Comment by bgirardot** [comment]
If I am reading things correctly, this pr should pass the travis ci check now that pgiraud has updated related files.

**Comment by bgirardot** [comment]
stackoverflow suggests @mikelmaron can trigger a travis ci rebuild by closing and then opening the pr. I think it will pass on the rebuild or at least not fail for flake8 checks.

http://stackoverflow.com/questions/17606874/trigger-a-travis-ci-rebuild-without-pushing-a-commit

**Comment by pgiraud** [comment]
Actually, I can also relaunch travis manually. Which I just did.

**Comment by bgirardot** [comment]
Oh cool, thank you very much @pgiraud Is there anything else we can do to get this one merged in?

**Comment by pgiraud** [comment]
It would be great to add tests. Just to maintain the coverage. This can be really trivial tests.

**Comment by bgirardot** [comment]
Ok, cool. I need to learn how to do that anyway :)

**Comment by mikelmaron** [comment]
@bgirardot are you taking this on now? I probably wouldn't have time to do so for a few days.

**Comment by bgirardot** [comment]
I can't say I can take this because I am not skilled enough in python/pyramid. I am however working on it and if I get lucky and get it figured out (adding to the data available via the api, adding the tests) I'll be happy to help with it much more. I might get these additional tests figured out today.

**Comment by bgirardot** [comment]
Hi @pgiraud and @mikelmaron I have written tests for this pull request here: bgirardot@SHA: 46467d30ddf21a33daff203eecc57d9768c0761f 

It covers all the additional missing statements nosetests reports. Not having written tests before, they basically just test the json is created more or less correctly and hopefully will not need updating if the json gets added to as long as the top level json structure doesn't change.

If they are ok, please merge in @mikelmaron 's PR and then I will make my PR, otherwise my PR will fail TravisCI if mikel's code is not in place for the tests to run on.

**Comment by pgiraud** [comment]
@mikelmaron can you please cherry-pick @bgirardot's commit in your branch?

**Comment by mikelmaron** [comment]
@pgiraud I directly edited my fork to add @bgirardot's changes. hope that does it for pushing live!

**Comment by bgirardot** [comment]
@pgiraud The failed travis ci build here is due to some larger problem with TM2 and its software stack I am pretty sure, nothing to do with this PR.

I made pull request #554 that does nothing essentially and it fails with the same travis ci errors and I can not get a fresh install of the current TM2 master branch to run properly locally either, mako errors, see issue #555

**Pull request merged**